pacai/agents/search/base.py

```diff
@@ -16,18 +19,29 @@ class SearchAgent(BaseAgent):
     """
 
     def __init__(self, index,
-            fn = 'pacai.student.search.depthFirstSearch',
```

Comment:
In all places where we are taking an optional function/class can be provided as an object or string, prefer an object.
This way, we will be able to take advantage of static type checking and automatically get alerted (in the form of failing code) if we ever change the name/package of these parameters.
Hopefully, this will ease your feeling of "hackiness".

For example, prefer:
```
fn = pacai.student.search.depthFirstSearch,
```
over
```
fn = 'pacai.student.search.depthFirstSearch',
```